client/main/vareditor/recodedvarwidget.js

```diff
@@ -4,26 +4,253 @@
 const $ = require('jquery');
 const Backbone = require('backbone');
 Backbone.$ = $;
+const keyboardJS = require('keyboardjs');
+const opsDropDown = require('../editors/operatordropdown');
 
 const RecodedVarWidget = Backbone.View.extend({
     className: 'RecodedVarWidget',
     initialize(args) {
 
         this.attached = false;
 
+        this._exampleFormulas = [
+            { s: ">", a: "2000", b: "'good'" },
+            { s: "<=", a: "1000", b: "A" },
+            { s: "==", a: "5", b: "B" },
+            { s: "<", a: "17000", b: "'Male'" },
+            { s: ">=", a: "1", b: "'Early'" },
+            { s: "=", a: "'tom'", b: "'medium'" }
+        ];
+
+        opsDropDown.init();
+
         this.$el.empty();
         this.$el.addClass('jmv-variable-recoded-widget');
+
+        this.$top = $('<div class="jmv-variable-recoded-top"></div>').appendTo(this.$el);
+        $('<div>recoded from variable</div>').appendTo(this.$top);
+        let t = '<select class="recoded-from">';
+        //for (let i = 0; i < options.length; i++)
+            t += '<option>A</option>';
+            t += '<option>B</option>';
+            t += '<option>C</option>';
```

Comment:
tidy up!

Reply:
fyi, the tidiness of the js is currently way off. I am just pushing over so that you are not hindered from doing the server side recode stuff. There is heaps more js code/clean up/re-do to come.